crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -16,6 +16,10 @@ use std::{
     sync::Arc,
 };
 
+/// An interaction paired with a flag indicating whether it can be omitted
+/// from the final execution plan
+type MaybeInternalizabelInteraction = (Arc<dyn Interaction>, bool);
```

Comment:
Have you considered defining another function on the trait Interaction that leaks whether it's private?
```
pub trait Intearction {
  fn encode() ->{};
  fn is_private_interaction(&self) -> bool
}
```
The advantage would be that we don't have to hand over tuples all the time. But I am not sure whether this abstraction is really well suited for all Interaction types and it could be too general. I have not thought deeply about it.

Reply:
Hey Felix, 
I did not approve yet, as I wanted someone else opinion
.
>My worry is that this logic would have to be implemented on every interaction (and someone adding a new interaction might forget to add support for internalizing)

If it's part of the trait the compile will tell them. I don't see this as a compelling reason.
But I think it's the right approach, given that we everywhere in the code keep the order information in the execution plans and they are separate from the interaction themselves.

Reply:
I think there is also a spelling error in the Name: MaybeInternalizabelInteraction 
Should it be MaybeInternalizableInteraction?

Maybe it's also wise to leave out the Maybe from the name, as the bool is strictly determining whether it's internalizable?

Reply:
Felix' reasoning makes sense to me. I think it is fine the way it is in the Pr now.